proto/storage/alert.proto

```diff
@@ -141,6 +147,9 @@ message Alert {
   ViolationState state = 11; // @gotags: search:"Violation State,store" sql:"index=btree"
   google.protobuf.Timestamp snooze_till = 12;
   reserved 14; // was tags
+
+  bool platform_component = 22; // @gotags: search:"Platform Component"
+  EntityType entity_type = 23;  // @gotags: search:"Entity Type"
```

Comment:
Could we not simply put `platform_component` in the `deployment` object and be done with it?  Is this because of how we poorly deal with `null` or because we may need this for `Resource` as well?

Reply:
I did not put `platform_component` directly in the deployment object because I assume there will soon be a time when we will want to mark alerts on Secrets, Config maps, etc as platform too if they belong to certain namespaces.

The entity type is for a Workloads tab will show only deployment type violations from non-platform deployment.